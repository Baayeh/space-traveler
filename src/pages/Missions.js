import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchMission,
  leaveMission,
  reserveMission,
} from '../redux/missions/missionSlice';

const Missons = () => {
  const { loading, missions, error } = useSelector((state) => state.mission);
  const dispatch = useDispatch();

  const joinMission = (id) => {
    dispatch(reserveMission(id));
  };

  const exitMission = (id) => {
    dispatch(leaveMission(id));
  };

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMission());
    }
  }, []);

  return (
    <div className="mission-wrapper">
      {loading && <div className="loading-msg">Loading...</div>}
      {!loading && error && <div>{error}</div>}
      {!loading && (
        <table className="mission-table">
          <thead className="thead-row">
            <tr>
              <th className="t-head">Mission</th>
              <th className="t-head">Description</th>
              <th className="t-head">State</th>
              <th className="t-head"> </th>
            </tr>
          </thead>
          <tbody>
            {missions?.map((mission) => (
              <tr key={mission.mission_id}>
                <td className="mission-name">{mission.mission_name}</td>
                <td className="td-desc">{mission.description}</td>
                <td className="td-state">
                  {!mission.reserved ? (
                    <span className="inactive-badge">Not A Member</span>
                  ) : (
                    <span className="active-badge">Active Member</span>
                  )}
                </td>
                <td className="td-state">
                  {!mission.reserved ? (
                    <button
                      type="button"
                      className="join-btn"
                      onClick={() => joinMission(mission.mission_id)}
                    >
                      Join Mission
                    </button>
                  ) : (
                    <button
                      type="button"
                      className="leave-btn"
                      onClick={() => exitMission(mission.mission_id)}
                    >
                      Leave Mission
                    </button>
                  )}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
};

export default Missons;
