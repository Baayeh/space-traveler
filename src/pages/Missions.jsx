import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import {
  fetchMission,
  leaveMission,
  reserveMission,
} from '../redux/missions/missionSlice';

const MySwal = withReactContent(Swal);

const Missions = () => {
  const { loading, missions, error } = useSelector((state) => state.mission);
  const dispatch = useDispatch();

  const joinMission = (id) => {
    dispatch(reserveMission(id));
    MySwal.fire({
      toast: true,
      position: 'top-end',
      timerProgressBar: true,
      icon: 'success',
      title: 'You have joined this mission',
      showConfirmButton: false,
      timer: 2000,
    });
  };

  const exitMission = (id) => {
    MySwal.fire({
      title: 'Are you sure you want to leave mission?',
      text: "You won't be able to revert this!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Yes, leave mission!',
      cancelButtonText: 'No, go back!',
      reverseButtons: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
    }).then((result) => {
      if (result.isConfirmed) {
        dispatch(leaveMission(id));
        MySwal.fire({
          toast: true,
          position: 'top-end',
          timerProgressBar: true,
          icon: 'success',
          title: 'You have left the mission',
          showConfirmButton: false,
          timer: 2000,
        });
      } else if (result.dismiss === MySwal.DismissReason.cancel) {
        MySwal.fire(
          'Cancelled!',
          'Action reverted!',
          'error',
        );
      }
    });
  };

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMission());
    }
  }, [dispatch, missions.length]);

  return (
    <section id="missions" className="p-[10rem] text-white">
      <div className="mission-wrapper">
        {loading && <div className="loading-msg">Loading...</div>}
        {!loading && error && <div>{error}</div>}
        {!loading && (
          <table className="mission-table mx-auto">
            <thead className="thead-row">
              <tr className="bg-slate-400 text-slate-800">
                <th className="border border-black p-4 w-[20rem] text-2xl uppercase">
                  Mission
                </th>
                <th className="border border-black p-4 w-[45rem] text-2xl uppercase">
                  Description
                </th>
                <th className="border border-black p-4 w-[15rem] text-2xl uppercase">
                  State
                </th>
                <th className="border border-black p-4 w-[15rem] text-2xl uppercase">
                  Action
                </th>
              </tr>
            </thead>
            <tbody>
              {missions?.map((mission) => (
                <tr key={mission.mission_id}>
                  <td className="mission-name border p-4 text-xl">
                    {mission.mission_name}
                  </td>
                  <td className="td-desc border p-4">{mission.description}</td>
                  <td className="td-state border p-4 text-center">
                    {!mission.reserved ? (
                      <span className="bg-slate-500 px-4 py-2 rounded-full">
                        Not A Member
                      </span>
                    ) : (
                      <span className="bg-green-600 px-4 py-2 rounded-full">
                        Active Member
                      </span>
                    )}
                  </td>
                  <td className="td-state border p-4 text-center">
                    {!mission.reserved ? (
                      <button
                        type="button"
                        className="border border-green-600 text-green-600 px-4 py-2 rounded-full hover:bg-green-600 hover:text-white transition-all duration-300 ease-in-out"
                        onClick={() => joinMission(mission.mission_id)}
                      >
                        Join Mission
                      </button>
                    ) : (
                      <button
                        type="button"
                        className="border border-red-600 text-red-600 px-4 py-2 rounded-full hover:bg-red-600 hover:text-white transition-all duration-300 ease-in-out"
                        onClick={() => exitMission(mission.mission_id)}
                      >
                        Leave Mission
                      </button>
                    )}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        )}
      </div>
    </section>
  );
};

export default Missions;
