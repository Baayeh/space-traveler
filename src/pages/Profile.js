import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { leaveMission } from '../redux/missions/missionSlice';
import { cancelRocket } from '../redux/rocket/rockets';

const Profile = () => {
  const { missions } = useSelector((state) => state.mission);
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  const exitMission = (id) => {
    dispatch(leaveMission(id));
  };

  const cancelReservation = (id) => {
    dispatch(cancelRocket(id));
  };

  const reservedMissions = missions.filter(
    (mission) => mission.reserved === true,
  );

  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <div className="profile-wrapper">
      <section className="mission-section">
        <h2>Missions</h2>
        <ul className="reserved-mission-list">
          {reservedMissions.length === 0 && <li>No missions reserved</li>}
          {reservedMissions
            && reservedMissions.map((mission) => (
              <li key={mission.mission_id}>
                <span>{mission.mission_name}</span>
                <div className="actions">
                  <button type="button" className="leave-btn" onClick={() => exitMission(mission.mission_id)}>
                    Leave Mission
                  </button>
                </div>
              </li>
            ))}
        </ul>
      </section>
      <section className="rockets-section">
        <h2>Rockets</h2>
        <ul className="reserved-mission-list">
          {reservedRockets.length === 0 && <li>No rockets reserved</li>}
          {reservedRockets
            && reservedRockets.map((rocket) => (
              <li key={rocket.id}>
                <span>{rocket.rockets_name}</span>
                <div className="actions">
                  <button type="button" className="leave-btn" onClick={() => cancelReservation(rocket.id)}>
                    Cancel Reservation
                  </button>
                </div>
              </li>
            ))}
        </ul>
      </section>
    </div>
  );
};

export default Profile;
