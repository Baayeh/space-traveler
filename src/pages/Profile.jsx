/* eslint-disable max-len */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import { leaveMission } from '../redux/missions/missionSlice';
import { cancelRocket } from '../redux/rocket/rockets';

const MySwal = withReactContent(Swal);

const Profile = () => {
  const { missions } = useSelector((state) => state.mission);
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  const exitMission = (id) => {
    MySwal.fire({
      title: 'Are you sure you want to leave mission?',
      text: "You won't be able to revert this!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Yes, leave mission!',
      cancelButtonText: 'No, go back!',
      reverseButtons: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
    }).then((result) => {
      if (result.isConfirmed) {
        dispatch(leaveMission(id));
        MySwal.fire({
          toast: true,
          position: 'top-end',
          timerProgressBar: true,
          icon: 'success',
          title: 'You have left the mission',
          showConfirmButton: false,
          timer: 2000,
        });
      } else if (result.dismiss === MySwal.DismissReason.cancel) {
        MySwal.fire(
          'Cancelled!',
          'Action reverted!',
          'error',
        );
      }
    });
  };

  const cancelReservation = (id) => {
    MySwal.fire({
      title: 'Are you sure you want to cancel?',
      text: "You won't be able to revert this!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Yes, cancel it!',
      cancelButtonText: 'No, go back!',
      reverseButtons: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
    }).then((result) => {
      if (result.isConfirmed) {
        dispatch(cancelRocket(id));
        MySwal.fire({
          toast: true,
          position: 'top-end',
          timerProgressBar: true,
          icon: 'success',
          title: 'Rocket Cancelled!',
          showConfirmButton: false,
          timer: 2000,
        });
      } else if (result.dismiss === MySwal.DismissReason.cancel) {
        MySwal.fire('Cancelled!', 'Action reverted!', 'error');
      }
    });
  };

  const reservedMissions = missions.filter(
    (mission) => mission.reserved === true,
  );

  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  return (
    <section id="profile" className="p-[10rem] text-white">
      <div className="profile-wrapper">
        <section className="mission-section">
          <h2 className="text-2xl mb-2">Missions</h2>
          <ul className="reserved-mission-list">
            {reservedMissions.length === 0 && <li>No missions reserved</li>}
            {reservedMissions
              && reservedMissions.map((mission) => (
                <li key={mission.mission_id}>
                  <span className="text-2xl">{mission.mission_name}</span>
                  <div className="actions">
                    <button
                      type="button"
                      className="border border-red-600 text-red-600 px-4 py-2 rounded-full hover:bg-red-600 hover:text-white transition-all duration-300 ease-in-out"
                      onClick={() => exitMission(mission.mission_id)}
                    >
                      Leave Mission
                    </button>
                  </div>
                </li>
              ))}
          </ul>
        </section>
        <section className="rockets-section">
          <h2 className="text-2xl mb-2">Rockets</h2>
          <ul className="reserved-mission-list">
            {reservedRockets.length === 0 && <li>No rockets reserved</li>}
            {reservedRockets
              && reservedRockets.map((rocket) => (
                <li key={rocket.id}>
                  <span className="text-2xl">{rocket.rockets_name}</span>
                  <div className="actions">
                    <button
                      type="button"
                      className="border border-red-600 text-red-600 px-4 py-2 rounded-full hover:bg-red-600 hover:text-white transition-all duration-300 ease-in-out"
                      onClick={() => cancelReservation(rocket.id)}
                    >
                      Cancel Reservation
                    </button>
                  </div>
                </li>
              ))}
          </ul>
        </section>
      </div>
    </section>
  );
};
export default Profile;
