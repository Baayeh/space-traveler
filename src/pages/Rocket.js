import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllRockets } from '../redux/rocket/rockets';

const Rocket = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchAllRockets());
  }, []);

  console.log(rockets);

  return (
    <div>
      <h1>All Rockets</h1>
    </div>
  );
};

export default Rocket;
