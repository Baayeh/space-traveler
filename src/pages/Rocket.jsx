import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import {
  cancelRocket,
  fetchAllRockets,
  reserveRocket,
} from '../redux/rocket/rockets';

const MySwal = withReactContent(Swal);

const Rocket = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  const bookRocket = (id) => {
    dispatch(reserveRocket(id));
    MySwal.fire({
      toast: true,
      position: 'top-end',
      timerProgressBar: true,
      icon: 'success',
      title: 'Rocket Booked!',
      showConfirmButton: false,
      timer: 2000,
    });
  };

  const cancelReservation = (id) => {
    MySwal.fire({
      title: 'Are you sure you want to cancel?',
      text: "You won't be able to revert this!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Yes, cancel it!',
      cancelButtonText: 'No, go back!',
      reverseButtons: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
    }).then((result) => {
      if (result.isConfirmed) {
        dispatch(cancelRocket(id));
        MySwal.fire({
          toast: true,
          position: 'top-end',
          timerProgressBar: true,
          icon: 'success',
          title: 'Rocket Cancelled!',
          showConfirmButton: false,
          timer: 2000,
        });
      } else if (result.dismiss === MySwal.DismissReason.cancel) {
        MySwal.fire(
          'Cancelled!',
          'Action reverted!',
          'error',
        );
      }
    });
  };

  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(fetchAllRockets());
    }
  }, [dispatch, rockets.length]);

  return (
    <section id="rockets" className="p-[10rem]">
      <div className="rocket-list text-white flex flex-wrap justify-center gap-7">
        {rockets
          && rockets?.map((rocket) => (
            <div
              key={rocket.id}
              className="rocket-card border rounded relative overflow-hidden"
            >
              {rocket && rocket.reserved === true && (
                <small className="bg-green-600 p-3 rounded-full absolute uppercase font-bold shadow-lg top-3 right-4">
                  Reserved
                </small>
              )}
              <div className="rocket-img rounded overflow-hidden w-[40rem] h-[30rem]">
                <img
                  src={rocket.flickr_images}
                  alt={rocket.rockets_name}
                  className="w-[40rem] h-[30rem]"
                />
              </div>
              <div className="rocket-overlay bg-orange-500">
                <div className="rocket-desc p-10 text-lg flex flex-col justify-center items-center h-full">
                  <h3 className="uppercase text-2xl font-bold underline">
                    {rocket.rockets_name}
                  </h3>
                  <p className="mt-1 mb-3 text-center font">
                    {rocket.description}
                  </p>
                  {rocket.reserved === true && (
                    <button
                      type="button"
                      className="border px-5 py-2 rounded-full hover:bg-white hover:text-orange-500 transition-all duration-300 ease-in-out"
                      onClick={() => cancelReservation(rocket.id)}
                    >
                      Cancel Reservation
                    </button>
                  )}
                  {rocket.reserved !== true && (
                    <button
                      type="button"
                      className="border px-5 py-2 rounded-full hover:bg-white hover:text-orange-500 transition-all duration-300 ease-in-out"
                      onClick={() => bookRocket(rocket.id)}
                    >
                      Reserve Rocket
                    </button>
                  )}
                </div>
              </div>
            </div>
          ))}
      </div>
    </section>
  );
};

export default Rocket;
