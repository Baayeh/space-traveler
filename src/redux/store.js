import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rocket/rockets';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
  },
});

export default store;
