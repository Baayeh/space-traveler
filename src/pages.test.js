import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from './redux/store';
import Rocket from './pages/Rocket';
import Missions from './pages/Missions';
import Profile from './pages/Profile';

describe('UI Test', () => {
  test('Test Profile UI', () => {
    const tree = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Test Mission UI', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Test Rocket UI', () => {
    const tree = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
