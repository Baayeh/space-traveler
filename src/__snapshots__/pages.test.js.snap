// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Test Test Mission UI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mission-wrapper"
      >
        <div
          class="loading-msg"
        >
          Loading...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mission-wrapper"
    >
      <div
        class="loading-msg"
      >
        Loading...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Test Test Profile UI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-wrapper"
      >
        <section
          class="mission-section"
        >
          <h2>
            Missions
          </h2>
          <ul
            class="reserved-mission-list"
          >
            <li>
              No missions reserved
            </li>
          </ul>
        </section>
        <section
          class="rockets-section"
        >
          <h2>
            Rockets
          </h2>
          <ul
            class="reserved-mission-list"
          >
            <li>
              No rockets reserved
            </li>
          </ul>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-wrapper"
    >
      <section
        class="mission-section"
      >
        <h2>
          Missions
        </h2>
        <ul
          class="reserved-mission-list"
        >
          <li>
            No missions reserved
          </li>
        </ul>
      </section>
      <section
        class="rockets-section"
      >
        <h2>
          Rockets
        </h2>
        <ul
          class="reserved-mission-list"
        >
          <li>
            No rockets reserved
          </li>
        </ul>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UI Test Test Rocket UI 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="rocket-list"
      />
    </div>
  </body>,
  "container": <div>
    <ul
      class="rocket-list"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
